codice/ddf
# Extract Solr from Catalog Core

#### What does this PR do?
Extracted all Solr concerns out of catalog core.
Extracted Solr cache from federation strategy in standard catalog framework.
Solr cache is now a source and plugins.
Removed unused, outdated, alternative federation strategies.

#### Who is reviewing it? 

@lamhuy 
@derekwilhelm 

#### Select relevant component teams: 

@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.

@rzwiefel
@lambeaux

#### How should this be tested?


- Install the standard install profile
- `feature:install catalog-solr-cache`
- Create loopback OpenSearch source named `loopback-os`
- Verify cache source is available
`curl --request GET \
  --url https://localhost:8993/services/catalog/sources \
  --header 'authorization: Basic YWRtaW46YWRtaW4=' \
  --header 'referer: https://localhost:8993/services/catalog/' \
  --header 'x-requested-with: XMLHttpRequest' \
  -k`
- Query loopback-os and request results to be cached
`curl --request GET \
  --url 'https://localhost:8993/services/catalog/query?q=*&src=loopback-os&mode=cache' \
  --header 'authorization: Basic YWRtaW46YWRtaW4=' \
  --header 'referer: https://localhost:8993/services/catalog/' \
  --header 'x-requested-with: XMLHttpRequest' \
  -k`
- Query cache to see loopback-os cached results
`curl --request GET \
  --url 'https://localhost:8993/services/catalog/query?q=*&src=cache' \
  --header 'authorization: Basic YWRtaW46YWRtaW4=' \
  --header 'referer: https://localhost:8993/services/catalog/' \
  --header 'x-requested-with: XMLHttpRequest' \
  -k`
- Clear cache from karaf shell `solr:cache --clear` and do previous command to verify cache results are 0

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: n/a

#### Screenshots


#### Checklist:
- [ n/a ] Documentation Updated
- [ n/a ] Update / Add Threat Dragon models

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
